docs/src/charm/howto/openstack.md

```diff
@@ -0,0 +1,120 @@
+# Integrating with OpenStack
+
+This guide explains how to integrate {{product}} with the OpenStack cloud
+platform.
+
+## Prerequisites
+
+To follow this guide, you will need:
+
+- An [OpenStack][openstack] cloud environment.
+- Juju set-up as the deployment tool for Openstack. Please refer to
+  [Juju and Openstack][juju-openstack] documentation for more information.
+- A Juju [controller][controller] with [access][credentials] to the OpenStack 
+  cloud environment.
+- A Juju [model][model] for deploying {{product}} on OpenStack.
+- A valid [proxy configuration][proxy] in constrained environments.
+
+## Deploying {{product}} on OpenStack
+
+
+To deploy the {{product}} [bundle][bundle] on OpenStack you need an overlay
+bundle which serves as an extension of the core bundle. The overlay bundle
+contains the necessary configuration to deploy {{product}} on OpenStack.
+Applications are deployed through the overlay and relations are established
+between the applications. These include the openstack integrator, cloud
+controller, and cinder-csi charm.
+
+Refer to the base overlay [openstack-overlay.yaml][openstack-overlay] and
+modify it as needed.
+
+### OpenStack Overlay Configurations:
+
+Run `openstack project list` to retrieve your project id and include the
+[project id][project] in the overlay template:
+
+```yaml
+applications:
+  openstack-integrator:
+    options:
+      project-id: <my-openstack-project-id>
+```
+
+
+Adjust [easyrsa][easyrsa] to avoid creating an LXD machine:
+
+```yaml
+applications:
+  easyrsa:
+    to:
+      - 0
+```
+
+If your set-up includes a load-balancer add the following:
+
+```yaml
+relations:
+  - [k8s:loadbalancer-external, openstack-integrator:lb-consumers]
+```
+
+```yaml
+applications:
+  kubeapi-load-balancer: null 
+```
```

Comment:
what's this?  I don't think our bundle call out the kubeapi-load-balancer does it?  This is used by charmed-kubernetes V1 -- it shouldn't be referenced here i don't think